test/mocks/lock.js

```diff
@@ -0,0 +1,42 @@
+var Fiber = require('fibers');
+
+var locked = false;
+var waitingFibers = [];
+
+var MockLock = function() {
+  this.gotLock = false;
+  this.id = Math.random();
+};
+
+MockLock.prototype.waitLock = function(timeoutInMillis) {
+  var start = new Date();
+  do {
+    if (!locked || this.gotLock) {
+      locked = true;
+      this.gotLock = true;
+      return;
+    } else {
+      waitingFibers.push(Fiber.current);
+      Fiber.yield();
+    }
+  } while (timeDiffInMillis(new Date(), start) < timeoutInMillis);
```

Comment:
This method is only used once. The *diff* can be expressed with a `-` (diff):
```js
do {
  ...
} while (new Date().getTime() - start.getMillis() < timeoutMillis);
```

Reply:
Done